src/plugins/kibana_usage_collection/server/collectors/ui_counters/register_ui_counters_collector.ts

```diff
@@ -41,7 +65,101 @@ export function transformRawCounter(rawUiCounter: UICounterSavedObject) {
     lastUpdatedAt,
     fromTimestamp,
     counterType,
-    total,
+    total: count,
+  };
+}
+
+export function transformRawUsageCounterObject(
+  rawUsageCounter: UsageCountersSavedObject
+): UiCounterEvent | undefined {
+  const {
+    id,
+    attributes: { count },
+    updated_at: lastUpdatedAt,
+  } = rawUsageCounter;
+  const { counterName, counterType, domainId } = deserializeCounterKey(id);
+
+  if (domainId !== 'uiCounter' || typeof count !== 'number' || count < 1) {
+    return;
+  }
+
+  const fromTimestamp = moment(lastUpdatedAt).utc().startOf('day').format();
+  const { appName, eventName } = deserializeUiCounterName(counterName);
+
+  return {
+    appName,
+    eventName,
+    lastUpdatedAt,
+    fromTimestamp,
+    counterType,
+    total: count,
+  };
+}
+
+export async function fetchUiCounters({ soClient }: CollectorFetchContext) {
+  const {
+    saved_objects: rawUsageCounters,
+  } = await soClient.find<UsageCountersSavedObjectAttributes>({
+    type: USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+    fields: ['count'],
+    perPage: 10000,
+  });
+
+  const { saved_objects: rawUiCounters } = await soClient.find<UICounterSavedObjectAttributes>({
+    type: UI_COUNTER_SAVED_OBJECT_TYPE,
+    fields: ['count'],
+    perPage: 10000,
+  });
```

Comment:
rollups in `ui_counters` will delete all the docs in the `ui-counter` saved object. so after a maximum of 3 days this repository will drain and rollups will still by completing the observer.

Once saved objects migrations enable cross SO repository migration we can do the delete this whole draining logic and simply move any remaining docs during migration.

Reply:
Right... it means once all of them are migrated, we will make this additional request knowing we'll get 0 docs. Should we consider stopping doing so at some point?

Either by changing the rollup to migrate to the new type of SOs (and we don't need to request this find) or by using `stopRollingUiCounterIndicies$` to identify when to stop requesting them? 🤔 

What do you think?

I recall having a similar discussion with @pgayvallet...

Reply:
Makes sense.

> Either by changing the rollup to migrate to the new type of SOs

Can you elaborate more on this?

Reply:
If we could use the UI Counter rollup method to migrate the `ui-counter` SavedObjects type, into the new `usage-counters` (read the `ui-counter` doc, store it as `usage-counters`, and delete the `ui-counter` doc), we don't need to query the `ui-counter` SOs neither any merging logic in the collector. What do you think?

Reply:
Syncing our discussion from slack:

I'm just worried that we'd be facing a race condition between the ui-counters `fetch` and the rollup. We're already working on migrations across registries so this would make deleting this code piece a lot easier when that lands.

I'm checking if the rxJs.subject is `complete` to stop querying the SO repository.